src/transformers/models/mobilebert/modeling_mobilebert.py

```diff
@@ -949,26 +949,13 @@ def __init__(self, config):
     def get_output_embeddings(self):
         return self.cls.predictions.decoder
 
-    def tie_weights(self):
-        """
-        Tie the weights between the input embeddings and the output embeddings. If the `torchscript` flag is set in the
-        configuration, can't handle parameter sharing so we are cloning the weights instead.
-        """
-        output_embeddings = self.get_output_embeddings()
-        input_embeddings = self.get_input_embeddings()
-
-        resized_dense = nn.Linear(
-            input_embeddings.num_embeddings, self.config.hidden_size - self.config.embedding_size, bias=False
+    def resize_token_embeddings(self, new_num_tokens: Optional[int] = None) -> torch.nn.Embedding:
+        # resize dense output embedings at first
+        self.cls.predictions.dense = self._get_resized_lm_head(
+            self.cls.predictions.dense, new_num_tokens=new_num_tokens, transposed=True
         )
-        kept_data = self.cls.predictions.dense.weight.data[
```

Comment:
this it's nice to refactor this out into `modeling_utils.py`. IMO, for the `resize_token_embeddings` function the user has to look into `modeling_utils.py` anyways and it's cleaner to have a consistent method `_get_resized_lm_head` with a `transposed` input parameter for resizing the lm head than having complex resize code here.

Is that ok for you @LysandreJik @sgugger ?

Reply:
SGTM